clients/go/pkg/commands/stream_jobs_test.go

```diff
@@ -0,0 +1,198 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"context"
+	"io"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/camunda/zeebe/clients/go/v8/internal/mock_pb"
+	"github.com/camunda/zeebe/clients/go/v8/internal/utils"
+	"github.com/camunda/zeebe/clients/go/v8/pkg/entities"
+	"github.com/camunda/zeebe/clients/go/v8/pkg/pb"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestStreamJobsCommand(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	client := mock_pb.NewMockGatewayClient(ctrl)
+	stream := mock_pb.NewMockGateway_StreamActivatedJobsClient(ctrl)
+
+	request := &pb.StreamActivatedJobsRequest{
+		Type:    "foo",
+		Timeout: DefaultJobTimeoutInMs,
+		Worker:  DefaultJobWorkerName,
+	}
+
+	job1 := pb.ActivatedJob{
+		Key:                      123,
+		Type:                     "foo",
+		Retries:                  3,
+		Deadline:                 123123,
+		Worker:                   DefaultJobWorkerName,
+		ElementInstanceKey:       123,
+		ProcessDefinitionKey:     124,
+		BpmnProcessId:            "fooProcess",
+		ProcessInstanceKey:       1233,
+		ElementId:                "foobar",
+		ProcessDefinitionVersion: 12345,
+		CustomHeaders:            "{\"foo\": \"bar\"}",
+		Variables:                "{\"foo\": \"bar\"}",
+	}
+	job2 := pb.ActivatedJob{
+		Key:           123,
+		Type:          "foo",
+		Retries:       3,
+		Deadline:      123123,
+		Worker:        DefaultJobWorkerName,
+		CustomHeaders: "{}",
+		Variables:     "{}",
+	}
+
+	expectedJobs := []entities.Job{{ActivatedJob: &job1}, {ActivatedJob: &job2}}
+
+	gomock.InOrder(
+		stream.EXPECT().Recv().Return(&job1, nil),
+		stream.EXPECT().Recv().Return(&job2, nil),
+		stream.EXPECT().Recv().Return(nil, io.EOF),
+	)
+
+	client.EXPECT().StreamActivatedJobs(gomock.Any(), &utils.RPCTestMsg{Msg: request}).Return(stream, nil)
+
+	ctx, cancel := context.WithTimeout(context.Background(), utils.DefaultTestTimeout)
+	defer cancel()
+
+	jobsChan := make(chan entities.Job, 2)
+	defer close(jobsChan)
+	err := NewStreamJobsCommand(client, func(context.Context, error) bool {
+		return false
+	}).JobType("foo").Consumer(jobsChan).Send(ctx)
+
+	assert.NoError(t, err)
+
+	// simulate job receive
+	jobs := []entities.Job{<-jobsChan, <-jobsChan}
+
+	if len(jobs) != len(expectedJobs) {
+		t.Error("Failed to receive all jobs: ", jobs, expectedJobs)
+	}
+
+	for i, job := range jobs {
+		if !reflect.DeepEqual(job, expectedJobs[i]) {
+			t.Error("Failed to receive job: ", job, expectedJobs[i])
+		}
+	}
+
+}
+
+func TestStreamJobsCommandWithTimeout(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	client := mock_pb.NewMockGatewayClient(ctrl)
+	stream := mock_pb.NewMockGateway_StreamActivatedJobsClient(ctrl)
+
+	request := &pb.StreamActivatedJobsRequest{
+		Type:    "foo",
+		Timeout: 60 * 1000,
```

Comment:
❓ Is the difference only this value?

Reply:
Yes, the mock verifies the request sent (i.e. built by the telescopic builder) is the one expected.